front/src/modules/companies/table/components/CompanyTable.tsx

```diff
@@ -32,14 +45,67 @@ export function CompanyTable() {
     return { AND: filters.map(turnFilterIntoWhereClause) };
   }, [filters]) as any;
 
+  const {
+    data: viewFieldsData,
+    loading: isViewFieldsLoading,
+    error: isViewFieldsError,
+  } = useGetViewFieldsQuery({
+    variables: { where: { objectName: { equals: 'company' } } },
+  });
+  const [createViewFieldsMutation] = useCreateViewFieldsMutation();
+
+  useEffect(() => {
+    if (
+      isViewFieldsLoading ||
+      isViewFieldsError ||
+      viewFieldsData?.viewFields.length
+    )
+      return;
+
+    createViewFieldsMutation({
+      variables: {
+        data: viewFieldDefinitions.map((viewFieldDefinition) => ({
+          fieldName: viewFieldDefinition.columnLabel,
+          index: viewFieldDefinition.columnOrder,
+          isVisible: true,
+          objectName: 'company',
+          sizeInPx: viewFieldDefinition.columnSize,
+        })),
+      },
+    });
+  }, [
+    createViewFieldsMutation,
+    isViewFieldsError,
+    isViewFieldsLoading,
+    viewFieldsData?.viewFields.length,
+  ]);
+
+  const viewFields = useMemo<ViewFieldDefinition<ViewFieldMetadata>[]>(
+    () =>
+      viewFieldsData?.viewFields.length
+        ? viewFieldsData.viewFields
+            .map((viewField) => ({
+              ...(viewFieldDefinitions.find(
+                ({ columnLabel }) => viewField.fieldName === columnLabel,
+              ) || { metadata: DEFAULT_VIEW_FIELD_METADATA }),
```

Comment:
OK, when we will serve this from the server we could remove this DEFAULT_VIEW_FIELD_METADATA, looks legit for now as we are not sure that what we have in FE matches what we have in BE